### ISSUE TITLE
[at_onboarding] [MacOS] Need a feature allow user share atSign between apps

### ISSUE THREAD

**Issue by sonle-geekyants** [atsign-foundation/at_client_sdk#1600]
In the new onboarding widget, we can give the user an option to share the the keychain between different apps. But this feature is only available in macos. 

We need to decide whether adding this feature only on macos will be fine or not. 
Currently this feature is disabled.

**Comment by nickelskevin** [comment]
IMHO - making use of keys that we have available to us to improve their experience is the preferred option.

**Comment by cconstab** [comment]
It would be my suggestion to make sure that the other desktops env all act the same . So that needs investigation. If they do this is a good idea.

 I think we should also look into allowing multiple instances of applications so they do no step on each others hive files..

**Comment by ksanty** [comment]
Leaving as a follow-up arch call discussion item

**Comment by nitesh2599** [comment]
@sonle-geekyants We will have to check a few things here :

Platform that supports share feature:

Desktop Platforms:
- [x] MacOS
- [x] Windows
- [ ] Linux

Mobile Platforms:
- [ ] Android
- [x] iOS (We will have to create ios app groups and test)

Data Related check:
- [x] We also need to make sure that if we use the share feature on a platform, two apps should not step on each others hive files .

**Comment by sachins-geekyants** [comment]
Reducing SP and moving to next sprint.

**Comment by ksanty** [comment]
Would like Gary's input when he's back

**Comment by sonle-geekyants** [comment]
@gkc and @sachins-geekyants 
Related to this issue:
- MacOS: ok
- iOS: ok. the app must add `Keychain Sharing` to `Capabilities` in XCode
- Android: don't support. `biometric_storage` use `KeyStore` in Android and can't share between apps
- Linux: don't support for now
- Windows: ok.

And don't make any impact to hive data if app use share atSign.

**Comment by nitesh2599** [comment]
Could not be worked on in this sprint, moving to next sprint.

**Comment by sonle-geekyants** [comment]
I've done this task and pushed code on branch 
- `feature/enable-user-sharing` of the Wavi app 
- `feature/enable-user-sharing` of the Atmosphere app
@nitesh2599 @sachins-geekyants @ayushkumarrojha you can test it on macOS, iOS and Windows.

**Comment by sachins-geekyants** [comment]
Reducing SP and moving to next sprint for testing and deployment.

**Comment by sachins-geekyants** [comment]
This could not be tested in the last sprint. Moving to next sprint.

@tinashe404 can you help us test following branch.
`feature/enable-user-sharing` of the Wavi app
`feature/enable-user-sharing` of the Atmosphere app

**Comment by tinashe404** [comment]
@sachins-geekyants sure lets have a chat about this when you're back on Wednesday.

**Comment by sonle-geekyants** [comment]
I've added a function `sharing atSign` on Menu Bar in the Wavi app and the Atmosphere app. Please test it @tinashe404 @ayushkumarrojha @sachins-geekyants 

https://user-images.githubusercontent.com/88494375/200545608-597881cf-73fe-4dfb-b0f8-985453a6997e.mov

**Comment by tinashe404** [comment]
I've tried this our on a simulator and have made the following observations

- If you open an app with an already onboarded atSign (in my case atmospherePro) and press start, it will ask if you want to share your atsign. At this point, i'm not sure which atSign is onboarded so i don't know if i want to share it or not. It would be good if we had something that shows which atSign it is. We also need to update the text on this screen, eg remove the @ in atPlatform. I will confirm wording tomorrow. 

- I pressed yes to allow sharing and then opened wavi. Wavi made me onboard and upload keys. I had assumed that if i had already said yes to sharing then if i opened wavi it would automatically onboard?

- What did work was if i added an atSign to one app, and then went to the other app and went to switch atsign, it would appear there too. The same for when i deleted it on one app, it would disappear from the other app under the switch atsign options. 

- if i choose to share an atSign, all other atSigns that i onboard after that are automatically shared too. should the default be on or off?

**Comment by sonle-geekyants** [comment]
> I had assumed that if i had already said yes to sharing then if i opened wavi it would automatically onboard?

-> Wavi will automatically onboard when you log in to HomePage -> enable sharing atSign before. 

> if i choose to share an atSign, all other atSigns that i onboard after that are automatically shared too. should the default be on or off?

-> Default should turn on. Because when allow sharing atSign, the app will share all data. If you don't want to share data, you shouldn't turn on sharing atSign.   
@tinashe404

**Comment by tinashe404** [comment]
Wording...

**Share this atSign with all your atPlatform apps?**

You will not be required to re-upload your atKeys when you use @tinashe404 with other atPlatform apps on this device.

**Comment by sachins-geekyants** [comment]
Moving to next sprint.

**Comment by sonle-geekyants** [comment]
Yes, I don't need to re-upload atKeys because when uploading atKeys in any app, I will save all information of that atSign in device. So I can share them with other atPlatform apps. @tinashe404

**Comment by sonle-geekyants** [comment]
@ayushkumarrojha please help me test this ticket.

**Comment by sachins-geekyants** [comment]
@sonle-geekyants @ayushkumarrojha can you update this ticket ?

**Comment by sonle-geekyants** [comment]
https://user-images.githubusercontent.com/88494375/210534986-7d0e571f-7595-4cbe-b67c-ce452e51a8c6.mov

I've uploaded the video @sachins-geekyants

**Comment by sachins-geekyants** [comment]
@sonle-geekyants Can we changes `@platform` to `atPlatform` in the message appears on dialog box ?

**Comment by sonle-geekyants** [comment]
> @sonle-geekyants Can we changes `@platform` to `atPlatform` in the message appears on dialog box ?

Okay. I will edit it.

**Comment by sonle-geekyants** [comment]
I've changed it @sachins-geekyants . Please pull the newest code and test it.
Branch:
The Wavi app: feature/enable-user-sharing
The atmospherePro app: feature/enable-user-sharing

https://user-images.githubusercontent.com/88494375/211784017-e51bc5b9-5993-43fb-bb47-88375d3d6b29.mov

**Comment by tinashe404** [comment]
@sachins-geekyants @sonle-geekyants the text still says " when you use @ sign with other apps"....

Shouldnt " @ sign" be the actual atSign that is currently onboarded?

**Comment by sachins-geekyants** [comment]
@tinashe404 This message appears before onboarding, so we don't know the atSign yet.

**Comment by tinashe404** [comment]
@sachins-geekyants in that case can we change the text to say "when you use this atSign with other"

**Comment by sonle-geekyants** [comment]
![image](https://user-images.githubusercontent.com/88494375/216214936-e43ffc56-362b-4024-acb0-2b038b286b86.png)

I've changed it. Could you please check again? @tinashe404

**Comment by tinashe404** [comment]
@sonle-geekyants looks good

**Comment by sachins-geekyants** [comment]
PRs have been raised for the above changes, I will assign it 0 SP till PR is reviewed.

**Comment by sachins-geekyants** [comment]
@sonle-geekyants can you check the status of PRs, it's still open.

**Comment by sachins-geekyants** [comment]
user sharing is available in `at_client_mobile 3.2.9`.
atApps needs to be deployed with options to enable sharing atSign info across apps.

**Comment by gkc** [comment]
@sachins-geekyants @ksanty  Is this still relevant, and something that needs to be worked on?

**Comment by sachins-geekyants** [comment]
The implementation is done in at_client_mobile and onboarding_flutter.

This feature is available in atPro desktop also, for other app we can create separate tickets if required.
This can be closed.

**Comment by ksanty** [comment]
> user sharing is available in `at_client_mobile 3.2.9`. atApps needs to be deployed with options to enable sharing atSign info across apps.

Is there a screen shot showing where we find this?

**Comment by sachins-geekyants** [comment]
> > user sharing is available in `at_client_mobile 3.2.9`. atApps needs to be deployed with options to enable sharing atSign info across apps.
> 
> Is there a screen shot showing where we find this?

This sharing atSign option

<img width="1434" alt="Screenshot 2024-03-19 at 4 54 23 PM" src="https://github.com/atsign-foundation/at_client_sdk/assets/74917871/5b32108b-02c7-402a-9187-65c1d30651fe">

**Comment by ksanty** [comment]
@sachins-geekyants can you provide guidance on when think this can be deployed?  TY

I think we also need to be mindful this feature is not supported in Android.
